guides/platformio.md

```diff
@@ -0,0 +1,137 @@
+---
+
+# {{ page.title }}
+
+
+Welcome to an introductory guide on taking the first steps with programming the STM32 microcontrollers within the PlatformIO ecosystem. The goal of this tutorial is to demonstrate how simple it is to set up PlatformIO with Visual Studio Code and use it to compile, upload, and debug STM32Cube-based application on the `STM32 Nucleo-F401RE` board.
+
+## What is PlatformIO?
+
+[PlatformIO](https://platformio.org/?utm_source=stm32-base) is a professional collaborative platform for embedded development that includes lots of instruments for the most common development tasks. It has out-of-the-box support for STM32 microcontrollers and allows working with low-level libraries like CMSIS, STM32Cube, and LibOpenCM3 as well as with popular high-level frameworks like Arduino, Mbed, or Zephyr without changing your usual development environment.
+
+A detailed overview of the PlatformIO ecosystem and its philosophy can be found in [the official documentation](https://docs.platformio.org/en/latest/what-is-platformio.html).
+
+## Installation
+
+Before diving into the development process, we need to install some required software. In this tutorial, we will use Visual Studio Code as the editor. The most up-to-date and detailed installation instructions can be found in the official [PlatformIO IDE for VSCode documentation](https://docs.platformio.org/en/latest/integration/ide/vscode.html#installation).
+
+Once the PlatformIO IDE is ready, we can install the [ST STM32 development platform](https://docs.platformio.org/en/latest/platforms/ststm32.html). This way PlatformIO will do all the heavy lifting for us by installing everything that is needed (toolchain, framework, etc) automatically. The easiest way to install any development platform is the PlatformIO Home App:
+
+![Getting Started with PlatformIO Installation]({{ site.url }}/assets/img/guides/platformio/platformio-install-1.png)
+{: style="text-align: center;"}
+
+When all installation steps are completed, a dialog window will pop up to tell you that the `ststm32` platform has been installed.
+
+## Creating a project
+
+In the first step, we need to create a new project using the same PlatformIO Home App:
+
+![Getting Started with PlatformIO Home]({{ site.url }}/assets/img/guides/platformio/setting-up-project-1.png)
+{: style="text-align: center;"}
+
+
+In the next step, we need to select `ST Nucleo-F401RE` as the development board, `STM32Cube` as the framework, and a path to the project location (or use the default one):
+
+![Getting Started with PlatformIO Project Init]({{ site.url }}/assets/img/guides/platformio/setting-up-project-2.png)
+{: style="text-align: center;"}
+
+##### Please note:
+
+Processing the selected project may take a considerable amount of time (PlatformIO will download and install all required packages) and after these steps, we have a fully configured project that is ready for developing code with the STM32Cube framework.
```

Comment:
`... amount of time as PlatformIO will download... ...all required packages. After these steps...`